treeherder/model/management/commands/import_perf_data.py

```diff
@@ -0,0 +1,112 @@
+                    default=False),
+        make_option('--num-workers',
+                    action='store',
+                    dest='num_workers',
+                    type='int',
+                    default=4),
+        make_option('--verbose',
+                    action='store_true',
+                    default=False),
+        make_option('--time-interval',
+                    action='store',
+                    default=None,
+                    type='int',
+                    help="Time interval to fetch (defaults to all)"),
+        make_option('--filter-props',
+                    action='append',
+                    dest="filter_props",
+                    help="Only import series matching filter criteria (can "
+                    "specify multiple times)",
+                    type="string",
+                    metavar="KEY:VALUE"),
+                )
+
+    def handle(self, *args, **options):
+        if len(args) != 1:
+            raise CommandError("Need to (only) specify project/branch")
+        project = args[0]
+
+        pc = PerfherderClient()
+        signatures = pc.get_performance_signatures(
+            project,
+            time_interval=PerformanceTimeInterval.NINETY_DAYS)
+
+        for kv in options['filter_props']:
+            if ':' not in kv or len(kv) < 3:
+                raise CommandError("Must specify --filter-props as "
+                                   "'key:value'")
+            k, v = kv.split(':')
+            signatures = signatures.filter((k, v))
+
+        if options['time_interval'] is None:
+            time_intervals = PerformanceTimeInterval.all_valid_time_intervals()
+        else:
+            time_intervals = [options['time_interval']]
+
+        with concurrent.futures.ProcessPoolExecutor(
+                options['num_workers']) as executor:
+            futures = []
+
+            for signature_hash in signatures.get_signature_hashes():
+                for time_interval in time_intervals:
+                    signature_props = signatures[signature_hash]
+                    futures.append(executor.submit(_add_series, project,
+                                                   time_interval,
+                                                   signature_hash,
+                                                   signature_props,
+                                                   options['mysql_debug'],
+                                                   options['verbose']))
+            for future in futures:
+                try:
+                    future.result()
+                except Exception, e:
+                    print "FAIL: %s" % e
+                    # shutdown any pending tasks and exit (if something
+                    # is in progress, no wait to stop it)
+                    executor.shutdown(wait=False)
+                    for future in futures:
+                        future.cancel()
+                    sys.exit(1)
```

Comment:
Maybe you can raise a CommandError here? In that way it can be caught when the command is executed using call_command